src/groups/mqb/mqbblp/mqbblp_queueengineutil.cpp

```diff
@@ -901,20 +901,20 @@ QueueEngineUtil_AppState::~QueueEngineUtil_AppState()
     // existing `RelayQueueEngine` routing contexts.
 }
 
-size_t QueueEngineUtil_AppState::catchUp(bsls::TimeInterval*          delay,
-                                         mqbi::StorageIterator*       reader,
-                                         mqbi::StorageIterator*       start,
-                                         const mqbi::StorageIterator* end)
+size_t QueueEngineUtil_AppState::catchUp(
+    bsls::TimeInterval*                                          delay,
+    mqbi::StorageIterator*                                       reader,
+    mqbi::StorageIterator*                                       start,
+    const bdlb::Variant<bsls::Types::Uint64, bmqt::MessageGUID>& stop)
```

Comment:
As discussed with @dorjesinpo, I will try to find a way to provide `stop` without building a `Variant` object. Template arg might work well, however, there might be cleaner ways to refactor the code